### ISSUE TITLE
Cannot install extensions in CRX format with any method

### ISSUE THREAD

**Issue by graphixillusion** [ungoogled-software/ungoogled-chromium-windows#23]
Hey there. I'm running the last version 78.0.3904.87. the option chrome://flags/#extension-mime-request-handling is set to Always prompt for install. When you try to install a crx, it hangs to: adding to chromium. I've tried with manually download and install via drang and drop but nothing happens. Previous version worked good.

**Comment by Eloston** [comment]
Identical problem posted here: https://github.com/Eloston/ungoogled-chromium/issues/834. Will leave this and the other issue open until the problem gets solved.

**Comment by Eloston** [comment]
I should also note that this doesn't work even with [Chromium Web Store](https://github.com/NeverDecaf/chromium-web-store), which suggests we can't install CRX files at all on Windows (however, Chromium Web Store does work on Linux).

**Comment by ianpan870102** [comment]
I faced the same issue earlier today. But after restarting my computer, it worked all right -- worth giving a try

**Comment by seanking2919** [comment]
Hi guys,

There's a piece of code in the function bool ExtensionManagement::IsOffstoreInstallAllowed under the file \src\chrome\browser\extensions\extension_management.cc that controls whether off-store installs are allowed or not by default. Here's what it looks like in Google's Chromium:
```
bool ExtensionManagement::IsOffstoreInstallAllowed(
    const GURL& url,
    const GURL& referrer_url) const {
  // No allowed install sites specified, disallow by default.
  if (!global_settings_->has_restricted_install_sources)
    return false;

  const URLPatternSet& url_patterns = global_settings_->install_sources;

  if (!url_patterns.MatchesURL(url))
    return false;

  // The referrer URL must also be whitelisted, unless the URL has the file
  // scheme (there's no referrer for those URLs).
  return url.SchemeIsFile() || url_patterns.MatchesURL(referrer_url);
}
```
Change the return false under if (!global_settings_->has_restricted_install_sources) to return true. Here's what the function should look like afterwards:
```
bool ExtensionManagement::IsOffstoreInstallAllowed(
    const GURL& url,
    const GURL& referrer_url) const {
  // No allowed install sites specified, allow by default.
  if (!global_settings_->has_restricted_install_sources)
    return true;

  const URLPatternSet& url_patterns = global_settings_->install_sources;

  if (!url_patterns.MatchesURL(url))
    return false;

  // The referrer URL must also be whitelisted, unless the URL has the file
  // scheme (there's no referrer for those URLs).
  return url.SchemeIsFile() || url_patterns.MatchesURL(referrer_url);
}
```

**Comment by seanking2919** [comment]
Hi,

Update: I tried editing the extension_management.cc file that way. Unfortunately, it didn't work.

**Comment by Eloston** [comment]
@seanking2919 Thanks for looking into this. There are modified builds of ungoogled-chromium on https://chromium.woolyss.com that exclude the Safe Browsing patches (in both this repo and ungoogled-chromium repo I think) and I've heard CRX installation works fine on them. You might have better luck looking through the Safe Browsing patches to see if anything looks fishy.

**Comment by seanking2919** [comment]
@Eloston Thank you for the information. Look at the patches that change chrome/browser/download/download_item_model.cc because when I looked at the patched file, none of the safe_browsing::DownloadFileType::DangerLevel stuff was in it.

**Comment by graphixillusion** [comment]
Ony my system the problem seems fixed since version 79. I can install extension as i always did.

**Comment by seanking2919** [comment]
@graphixillusion Specific version you're currently using?

**Comment by graphixillusion** [comment]
@seanking2919 version 80.0.3987.163 x64

**Comment by seanking2919** [comment]
Ok. @graphixillusion And out of pure curiosity, does the build use the "Safe Browsing" patches? Or did you get the build from Woolyss?

**Comment by graphixillusion** [comment]
@seanking2919 i use chrlauncher (https://github.com/henrypp/chrlauncher) for getting the build.

**Comment by seanking2919** [comment]
I see. So, basically, the builds without the Safe Browser patches.

**Comment by graphixillusion** [comment]
@seanking2919  how to verify if the Sage Browser patches are presents or not?

**Comment by seanking2919** [comment]
If you got it with chrlauncher, there's a good chance it does not have the Safe browser patches.

**Comment by Eloston** [comment]
Should be solved in 81.0.4044.138-1.1
